### PR TITLE
Adding backend pagination

### PR DESCRIPTION
**Pull request by GPaoloni**
Jira: https://bugs.benetech.org/browse/CHI-179
This PR is related to https://github.com/tech-matters/flex-plugins/pull/135

This PR adds backend pagination for the cases list.

### CODE REVIEW

File: controllers/case-controller.js [hunk 1]
```diff
@@ -26,16 +26,20 @@ const CaseController = Case => {
   };
 
   const listCases = async query => {
+    const limit = query.limit && parseInt(query.limit, 10);
```
Comment (nickhurlburt) [inline controllers/case-controller.js:29]: One principle of API design that I've learned the hard way is that we should always have limits.  A 'rogue' client should never be able to take us down by asking for, say, one million results.  So maybe while we're here, we make this something like `const limit = parseInt(query.limit, 10) || 1000;` ?

File: controllers/case-controller.js [hunk 2]
```diff
@@ -26,16 +26,20 @@ const CaseController = Case => {
...
         helpline: query.helpline || '',
       },
+      limit,
+      offset,
     };
 
-    const cases = await Case.findAll(queryObject);
-    const withContactInfo = await Promise.all(
-      cases.map(async caseItem => {
+    const { count, rows } = await Case.findAndCountAll(queryObject);
```
Comment (nickhurlburt) [inline controllers/case-controller.js:40]: There's a `findAndCountAll`!  That's awesome.  Though it's good to keep in mind that behind the scenes, sequelize is probably making at least two queries.

Comment (GPaoloni) [reply controllers/case-controller.js:40]: Yes, in fact it does, you can check that in the console after the function is called. Should I leave some comment around this?

File: controllers/case-controller.js (hunk 1 above)
Comment (GPaoloni) [reply controllers/case-controller.js:29]: Right! Should we choose the minimum between `query.limit` and 1000, or whatever number the user sends in the query will be fine?

**Pull request merged**